src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzIamStep.java

```diff
@@ -50,6 +52,18 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
           SnapshotDuosFlightUtils.getFirecloudGroup(context);
       derivedPolicies.addReadersItem(duosFirecloudGroup.getFirecloudGroupEmail());
     }
+    if (snapshotRequestModel.getContents().get(0).getMode()
+        == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
```

Comment:
why does this happen in this step and not a separate step?

Reply:
By adding it to this step, we can include it in the call to sam to create the request. If it was in a separate step, we would have to make another call to SAM (as opposed to being able to group it in with the create request).
Additionally, it is following the DUOS group model (We add the DUO firecloud/sam group right above this call).